vscode/src/chat/GraphContextProvider.ts

```diff
@@ -304,20 +304,38 @@ export const gatherDefinitions = async (
             continue
         }
 
+        const requestQueue: { symbolName: string; position: vscode.Position }[] = []
         for (const { start, end } of [range]) {
             for (const [lineIndex, line] of lines.slice(start.line, end.line + 1).entries()) {
-                for (const match of line.matchAll(identifierPattern)) {
+                // NOTE: pretty hacky - strip out C-style line comments and find everything
+                // that might look like it could be an identifier. If we end up running a
+                // VSCode provider over this cursor position and it's not a symbol we can
+                // use, we'll just get back an empty location list.
+                const identifierMatches = line.replace(/\/\/.*$/, '').matchAll(identifierPattern)
+
+                for (const match of identifierMatches) {
                     if (match.index === undefined || commonKeywords.has(match[0])) {
                         continue
                     }
 
-                    definitionMatches.push({
+                    requestQueue.push({
                         symbolName: match[0],
-                        locations: getDefinitions(uri, new vscode.Position(start.line + lineIndex, match.index + 1)),
+                        position: new vscode.Position(start.line + lineIndex, match.index + 1),
                     })
                 }
             }
         }
+
+        // NOTE: deduplicating here will save duplicate queries that are _likely_ to point to the
+        // same definition, but we may be culling aggressively here for some edge cases. I don't
+        // currently think that these are likely to be make-or-break a quality response on any
+        // significant segment of real world questions, though.
+        for (const { symbolName, position } of dedupeWith(requestQueue, ({ symbolName }) => symbolName)) {
```

Comment:
This is also possible now with type-safety: 

```suggestion
        for (const { symbolName, position } of dedupeWith(requestQueue, 'symbolName')) {
```

Reply:
WOAH. Can you explain this?

Reply:
I wanted to use this util too, so I moved it to the `shared` package and tweaked the implementation. The second argument now accepts `keyof T`, which is used if `typeof key !== 'function'`. 

https://github.com/sourcegraph/cody/blob/c121137003ec58c050b8eb16ccd6356693d04af6/lib/shared/src/common/index.ts#L60-L62

Reply:
Updated!